.azure-pipelines/pipelines.yml

```diff
@@ -0,0 +1,60 @@
+trigger:
+  branches:
+    include:
+    - "main"
+
+stages:
+- stage: check
+  dependsOn: []
+  pool: "x64-large"
+  jobs:
+  - job: build_and_format
+    displayName: "do_ci.sh"
+    dependsOn: []
+    strategy:
+      maxParallel: 2
+      matrix:
+        build:
+          CI_TARGET: "build"
```

Comment:
Correct me if I am wrong, but I suspect Salvo doesn't need a lot of resources to run the `build`, `check_format`, `test` and `coverage` stages. The way this is set up now, each one of them will run on a separate VM.

Can we define them all in a single stage running them as parallel jobs? Same as the build` and `check_format` here.

Reply:
Yes, I agreed. I will move them into one single stage.